src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -60,7 +61,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
             requestPage = "product.jsp";
             Cart cartFromSession = cartService.getCart(request);
             request.setAttribute("cart", cartFromSession);
-            productDao.addProductToRecentlyViewed(request, product);
+            new RecentlyViewedProductsService().addProductToRecentlyViewed(request, product);
```

Comment:
Create field RecentlyViewedProductsService in Servlet class as productDao and CartService

Reply:
Ok